webapp/channels/src/components/new_search/search_box_suggestions.tsx

```diff
@@ -0,0 +1,116 @@
+    font-size: 12px;
+    line-height: 16px;
+    font-weight: 600;
+    text-transform: uppercase;
+`;
+
+const SuggestionsBody = styled.div`
+    border-bottom: 1px solid rgba(var(--center-channel-color-rgb), 0.08);
+    padding-bottom: 16px;
+`;
+
+type Props = {
+    searchType: string;
+    searchTerms: string;
+    setSearchTerms: (searchTerms: string) => void;
+    selectedOption: number;
+    setSelectedOption: (idx: number) => void;
+    suggestionsHeader: React.ReactNode;
+    providerResults: ProviderResult<unknown>|null;
+    focus: (newPosition: number) => void;
+    onSearch: (searchType: string, searchTerms: string) => void;
+}
+
+const SearchSuggestions = ({searchType, searchTerms, setSearchTerms, suggestionsHeader, providerResults, selectedOption, setSelectedOption, focus, onSearch}: Props) => {
+    const license = useSelector(getLicense);
+    const updateSearchValue = useCallback((value: string, matchedPretext: string) => {
+        const changedValue = value.replace(matchedPretext, '');
+        setSearchTerms(searchTerms + changedValue + ' ');
+        focus(searchTerms.length + changedValue.length + 1);
+    }, [searchTerms, setSearchTerms, focus]);
+
+    const runSearch = useCallback((searchTerms: string) => {
+        onSearch(searchType, searchTerms);
+    }, [onSearch, searchType]);
+
+    let SearchPluginSuggestions = useSelector((state: GlobalState) => state.plugins.components.SearchSuggestions) || [];
+    if (license.IsLicensed !== 'true') {
+        SearchPluginSuggestions = [];
+    }
+
+    if ((searchType === '' || searchType === 'messages' || searchType === 'files') && providerResults) {
+        return (
+            <SuggestionsBody>
+                <SuggestionsHeader>{suggestionsHeader}</SuggestionsHeader>
+                {providerResults.items.map((item, idx) => {
+                    if (!providerResults.component) {
+                        return null;
+                    }
+                    const Component = providerResults.component as React.ComponentType<SuggestionProps<any>>;
+                    return (
+                        <Component
+                            key={providerResults.terms[idx]}
+                            item={item as UserProfile}
+                            term={providerResults.terms[idx]}
+                            matchedPretext={providerResults.matchedPretext}
+                            isSelection={idx === selectedOption}
+                            onClick={updateSearchValue}
+                            onMouseMove={() => {
+                                setSelectedOption(idx);
+                            }}
+                        />
+                    );
+                })}
+            </SuggestionsBody>
+        );
+    }
+
+    const pluginComponentInfo = SearchPluginSuggestions.find(({pluginId}: any) => {
+        if (searchType === pluginId) {
+            return true;
+        }
+        return false;
+    });
```

Comment:
0/5 Probably early optimization, but depending on the potential size of `SearchPluginSuggestions` it may be reasonable to add a `useMemo` here.

Reply:
Yes, it sounds like an early optimization to me. There is going to be normally 0 or 1 plugin there, so I don't think there is much value on memorizing that.